From 4272dc4bfd1c2945e00c82c27760a3ae6eed28dc Mon Sep 17 00:00:00 2001
From: Michal Srb <msrb@redhat.com>
Date: Wed, 24 Sep 2014 16:48:57 +0200
Subject: [PATCH] [metadata] Read OSGi Requires from manifest only if osgi.id
 property is missing in metadata

Missing osgi.requires key in properties doesn't mean that we should
always try to read Requires from manifest manually. XMvn skips this
property, if there are no interesting Requires.
---
 python/javapackages/metadata/metadata.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/python/javapackages/metadata/metadata.py b/python/javapackages/metadata/metadata.py
index b7c4c9f..5b507e3 100644
--- a/python/javapackages/metadata/metadata.py
+++ b/python/javapackages/metadata/metadata.py
@@ -200,7 +200,13 @@ class Metadata(object):
                             reqs |= set(content.split(','))
                             continue
                         except KeyError:
-                            pass
+                            try:
+                                osgi_id = artifact.properties["osgi.id"]
+                                # this file was already processed by XMvn and
+                                # there are no interesting OSGi requires, move on
+                                continue
+                            except KeyError:
+                                pass
                     if artifact.path:
                         import javapackages.common.osgi as osgi
                         r = osgi.get_requires(artifact.get_buildroot_path())
-- 
1.9.3

